packages/grid/_modules_/grid/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -118,3 +119,40 @@ export const buildAggregatedFilterApplier = (
     return filteredAppliers.some((applier) => applier.fn(rowId));
   };
 };
+
+const filterModelDisableMultiColumnsFilteringWarning = buildWarning([
+  'MUI: The `filterModel` can only contain a single item when `prop.disableMultipleColumnsFiltering` is set to `true`.',
+  'If you are using the `DataGrid` community version, this property is always `true`.',
+]);
+
+const filterModelMissingItemIdWarning = buildWarning(
+  "MUI: The 'id' field is required on `filterModel.items` when you use multiple filters.",
+);
+
+export const checkFilterModelValidity = (
+  model: GridFilterModel,
+  disableMultipleColumnsFiltering: boolean,
+) => {
+  if (model.items.length > 1) {
+    if (disableMultipleColumnsFiltering) {
+      filterModelDisableMultiColumnsFilteringWarning();
+
+      return {
+        ...model,
+        items: [model.items[0]],
+      };
+    }
+
+    const hasItemsWithoutIds = model.items.find((item) => item.id == null);
+    if (hasItemsWithoutIds) {
+      filterModelMissingItemIdWarning();
+
+      return {
+        ...model,
+        items: [model.items[0]],
```

Comment:
You could either let items as they are (without id) or generate ids such as `item.id = "youForgetToSetThisId-1"` for example.
The solution before this commit was to only send a warning

Reply:
I just realized missing id are filled in `cleanFilterItem`

https://github.com/alexfauquette/material-ui-x/blob/master/packages/grid/_modules_/grid/hooks/features/filter/gridFilterUtils.ts#L25:L27

Reply:
Yes when using `upsertFilterItem` or `showFilterPanel` we allow the input item to lack the id.
But when the model is created by the user and passed as a prop or directly to `setFilterModel`, the id is required.

Reply:
Why not using the same behavior in `setFilterModel`? 

Like for the `key` prop in react to render an array. As a developer, if I don't set `key` I expose myself to bugs and I see warnings in the console. But in most cases, from a user perspective, my code runs as expected. Here the code says set multiple filters, and the user will only see one filter

Reply:
Because when controlled we want the state to reflect what the user gives, no matter how invalid it is.
We could revert the logic and do the cleaning in a selector like we do for the column grouping.

Reply:
If the state must reflect what the developer provides, it would make sense to have
```diff
-return {
- 	...model,
-     items: [model.items[0]],
- };
+return model;
```
About adding the ids in the selector, I don't see how you could do it in such a way that it would be consistent between two calls

Reply:
You are right, I mixed several things...
It would be nice to have what I described but it clearly requires some deeper changes.
We could add the id here indeed.